### PR TITLE
Feature/get ball tactic

### PR DESCRIPTION
**Pull request by Ratoone**
Tactic for GetBall. See #869. It computes the closest point to the ball from the robot, drives towards it, faces the ball and starts the dribbler

### CODE REVIEW

File: src/stp/new_tactics/GetBall.cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+        StpInfo tacticInfo = info;
+        Vector2 robotPosition = info.getRobot().value()->getPos();
+        Vector2 ballPosition = info.getBall().value()->getPos();
+
+        // the robot will go to the position of the ball
+        double ballDistance = (ballPosition - robotPosition).length();
+        Vector2 newRobotPosition = robotPosition + (ballPosition - robotPosition).stretchToLength(ballDistance - Constants::ROBOT_RADIUS());
+        if (ballDistance < 3 * Constants::ROBOT_RADIUS()){
+            tacticInfo.setAngle((float)(ballPosition - robotPosition).angle());
```
Comment (jordiluong) [inline src/stp/new_tactics/GetBall.cpp]:
```suggestion
            tacticInfo.setAngle((ballPosition - robotPosition).angle());
```
is double now

File: src/stp/new_tactics/GetBall.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            tacticInfo.setDribblerSpeed(31);
```
Comment (jordiluong) [inline src/stp/new_tactics/GetBall.cpp]:
```suggestion
            tacticInfo.setDribblerSpeed(100);
```
dribblerspeed is set to 0% - 100% in StpInfo now. The skill converts it to robot command

File: src/stp/new_tactics/GetBall.cpp [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        }
+
+        tacticInfo.setTargetPos(std::make_pair(targetType::MOVETARGET, newRobotPosition));
```
Comment (jordiluong) [inline src/stp/new_tactics/GetBall.cpp]:
```suggestion
        tacticInfo.setPosition(std::make_pair(PositionType::MOVE_TO_POSITION, newRobotPosition));
```
New namings

File: src/stp/new_tactics/GetBall.cpp [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+    void GetBall::onUpdate(Status const &status) noexcept {}
+
+    void GetBall::onTerminate() noexcept {}
+
+    StpInfo GetBall::calculateInfoForSkill(StpInfo const &info) noexcept {
+        if (!info.getBall() || !info.getRobot()) {
+            RTT_WARNING("No Ball or Robot present in StpInfo");
+            return {};
+        }
```
Comment (timovdk) [inline src/stp/new_tactics/GetBall.cpp]: Nice! I think it's smart to put this kind of check in all calculateInfos in all tactics, maybe also check for field?

File: src/stp/new_tactics/GetBall.cpp [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+        StpInfo tacticInfo = info;
```
Comment (timovdk) [inline src/stp/new_tactics/GetBall.cpp]: Maybe skillInfo instead of tacticInfo? since the info generated here is specifically made for the skills

**Review by timovdk — APPROVED** [review]

**Pull request merged**